navigation/src/main/java/no/nordicsemi/android/common/navigation/Navigator.kt

```diff
@@ -0,0 +1,53 @@
+package no.nordicsemi.android.common.navigation
+
+import android.net.Uri
+import android.os.Bundle
+import androidx.lifecycle.SavedStateHandle
+
+interface Navigator {
+    /**
+     * Requests navigation using the given argument.
+     *
+     * The destination target is determined either by the [Router] for the
+     * local [NavigationDestinations], or the global [Router] set in
+     * [NavigationView].
+     *
+     * @param hint An optional hint to determine the target destination.
+     * @param args An optional argument to pass to the destination.
+     */
+    fun navigate(hint: Any? = null, args: Bundle? = null)
+
+    /**
+     * Requests navigation using the given argument and suspends until the result is received.
+     *
+     * The destination target is determined either by the [Router] for the
+     * local [NavigationDestinations], or the global [Router] set in
+     * [NavigationView].
+     *
+     * The target destination must call [navigateUpWithResult] to return the result,
+     * or [navigateUp] to cancel. If cancelled, null will be returned.
+     *
+     * @param hint An optional hint to determine the target destination.
+     * @param args An optional argument to pass to the destination.
+     */
+    suspend fun <T> navigateForResult(hint: Any? = null, args: Bundle? = null): T?
```

Comment:
Very nice implementation, but I am wondering if we were too enthusiastic about this. 

When the application is killed and recreated, the lunching coroutine will be stopped. Unless it happens in ViewModel's init() method it will not work. Can you verify this scenario?

Reply:
As always, you were right. I split it into subscribe + navigate.
The new implementation was tested with *Don't keep Activities* and seems to be working.